app/classifier/tasks/survey/index.jsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Editor from './editor';
+import Summary from './summary';
+import Chooser from './chooser';
+import Choice from './choice';
+import AnnotationView from './annotation-view';
+
+export default class SurveyTask extends React.Component {
+  static Editor = Editor
+
+  static Summary = Summary
+
+  static AfterSubject = AnnotationView
+
+  static getDefaultTask() {
+    return {
+      type: 'survey',
+      characteristicsOrder: [],
+      characteristics: {},
+      choicesOrder: [],
+      choices: {},
+      questionsOrder: [],
+      questions: {},
+      images: {}
+    };
+  }
+
+  static getTaskText(task) {
+    return `Survey of ${task.choicesOrder.length} choices`;
+  }
+
+  static getDefaultAnnotation() {
+    return { value: [] };
+  }
+
+  static isAnnotationComplete(task, annotation) {
+    const minRequiredAnswers = task.required ? 1 : 0;
+    return annotation.value.length >= minRequiredAnswers && !annotation._choiceInProgress;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.clearSelection = this.clearSelection.bind(this);
+    this.handleAnnotation = this.handleAnnotation.bind(this);
+    this.handleChoice = this.handleChoice.bind(this);
+    this.handleFilter = this.handleFilter.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+
+    this.state = {
+      filters: {},
+      focusedChoice: '',
+      selectedChoiceID: ''
+    };
+  }
+
+  handleFilter(characteristicID, valueID) {
+    const { filters } = this.state;
+
+    setTimeout(() => {
```

Comment:
Does the survey task break if you remove this timeout? It's hard to see why it's here, and it might not be needed.

Reply:
Currently, when you select a filter, the modal asking which filter you want closes after making a selection. If `setTimeout` is removed, the modal doesn't close after making a selection, you then have to click off of the modal.